examples/libxml2/libxml2_example.cc

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "examples/libxml2/xml_writer.h"
+#include "libxml/parser.h"
+#include "src/libfuzzer_protobuf_mutator.h"
+#include "xml.pb.h"  // NOLINT
+
+using examples::libxml2::Xml;
+
+namespace {
+google::protobuf::LogSilencer log_silincer;
+}
+
+extern "C" size_t LLVMFuzzerCustomMutator(uint8_t* data, size_t size,
+                                          size_t max_size, unsigned int seed) {
+  Xml message;
+  // Fuzzer can be provided with corpus of raw XML. In this case it will fail
+  // to parse it as proto.
+  bool is_proto = protobuf_mutator::ParseTextMessage(data, size, &message);
+  // If the data is a raw XML we can store it as a content field of the proto.
+  // Also we can do the same as a special kind of mutation. If the data is a
+  // proto we can convert it to XML and store as a content field. This mutation
+  // allows to grow coverage faster.
+  if (!is_proto || seed % 33 == 0) {
```

Comment:
Not sure. It's quite a hassle to do preprocessing instead of "fuzzer corpus/ maybe_interesting_stuff/" Where maybe_interesting_stuff what ever you want.
Also case for (seed % 33 == 0) very helpful to increase coverage.